pkg/entity/known_ids.go

```diff
@@ -17,6 +20,7 @@ const defaultTTL = 24 * time.Hour
 //
 // This component is not thread-safe.
 type KnownIDs struct {
+	lock sync.Mutex
```

Comment:
Might be better to replace this with a [`sync.RWMutex`](https://golang.org/pkg/sync/#RWMutex) as it will allow multiple reads and only lock for an update.

Reply:
Done, thanks!!!